rest/src/main/java/com/redhat/ipaas/rest/v1/util/ResultFilter.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.rest.v1.util;
+
+
+import com.redhat.ipaas.model.ListResult;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Generic filter that removes items that do not match predicate
+ */
+public class ResultFilter<T> implements Function<ListResult<T>, ListResult<T>> {
```

Comment:
Would call it `PredicateFilter` since `ResultFilter` is a bit unspecific (the `PaginationFilter` is also a `ResultFilter`)